apps/examples/testcase/le_tc/kernel/itc_pthread.c

```diff
@@ -0,0 +1,505 @@
+	ret_chk = pthread_cond_broadcast(&cond_attrparam);
+	TC_ASSERT_EQ("pthread_cond_broadcast", ret_chk, EINVAL);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_pthread_getname_np_p_defaultname
+* @brief                :The pthread_getname_np function gets the name of pthread
+* @Scenario             :create pthread and get its default name
+* API's covered         :pthread_setname_np, pthread_getname_np
+* Preconditions         :none
+* Postconditions        :none
+*/
+static void itc_pthread_getname_np_p_defaultname(void)
+{
+	int ret_chk;
+	pthread_t name_th;
+	char get_name[BUF_LEN];
+	char *default_name = "<pthread>";
+
+	ret_chk = pthread_create(&name_th, NULL, thread_setgetname, NULL);
+	TC_ASSERT_EQ("pthread_create", ret_chk, OK);
+
+	ret_chk = pthread_getname_np(name_th, get_name);
+	TC_ASSERT_EQ("pthread_getname_np", ret_chk, OK);
+	TC_ASSERT_EQ("pthread_getname_np", strcmp(get_name, default_name), 0);
+
+	ret_chk = pthread_join(name_th, NULL);
+	TC_ASSERT_EQ("pthread_join", ret_chk, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_pthread_setgetname_np_p_reset_name
+* @brief                :The pthread_getname_np function gets the name of pthread
+* @Scenario             :create pthread and get its default name
+* API's covered         :pthread_setname_np, pthread_getname_np
+* Preconditions         :none
+* Postconditions        :none
+*/
+static void itc_pthread_setgetname_np_p_reset_name(void)
+{
+	int ret_chk;
+	int exec_index;
+	int len;
+	pthread_t name_th;
+	char thread_name[BUF_LEN];
+	char *thread_str = "NameThread";
+	char get_name[BUF_LEN];
+
+	len = strlen(thread_name);
+	ret_chk = pthread_create(&name_th, NULL, thread_setgetname, NULL);
+	TC_ASSERT_EQ("pthread_create", ret_chk, OK);
+
+	for (exec_index = 0; exec_index < LOOP_SIZE; exec_index++) {
+		snprintf(thread_name, BUF_LEN, "%s_%d", thread_str, exec_index);
+		ret_chk = pthread_setname_np(name_th, thread_name);
+		TC_ASSERT_EQ_CLEANUP("pthread_setname_np", ret_chk, OK, pthread_join(name_th, NULL));
+
+		ret_chk = pthread_getname_np(name_th, get_name);
+		TC_ASSERT_EQ_CLEANUP("pthread_getname_np", ret_chk, OK, pthread_join(name_th, NULL));
+		TC_ASSERT_EQ_CLEANUP("pthread_getname_np", strcmp(get_name, thread_name), 0, pthread_join(name_th, NULL));
+	}
+
+	ret_chk = pthread_join(name_th, NULL);
+	TC_ASSERT_EQ("pthread_join", ret_chk, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_pthread_equal_pthread_self_p
+* @brief                :ompares two thread which is equal or not
+* @Scenario             :create 2 pthread and compares two thread
+* API's covered         :pthread_equal, pthread_self
+* Preconditions         :none
+* Postconditions        :none
+*/
+static void itc_pthread_equal_pthread_self_p(void)
+{
+	int ret_chk;
+
+	g_flag = false;
+	ret_chk = pthread_create(&g_pthread, NULL, thread_equal_check, NULL);
+	TC_ASSERT_EQ("pthread_create", ret_chk, OK);
+	TC_ASSERT_EQ("pthread_create", g_flag, true);
```

Comment:
New pthread attr is NULL. That means new one has a same priority with parent. You can't check this flag like this.

Reply:
This comment has been fixed.